atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java

```diff
@@ -282,16 +282,21 @@ private void executeHandlingExceptions(ThrowingVoidSupplier supplier)
     }
 
     private void updateIsValid(Exception e) {
```

Comment:
Was wondering if we need to `synchronise` this method.

Turns out we don't, since `compareAndSet` simply returns false if we've lost a race with another thread, and attempting to invalidate something twice has the same result as attempting it once.